mlonmcu/flow/backend.py

```diff
@@ -72,8 +74,26 @@ def load_model(self, model):
         pass
 
     @abstractmethod
+    def _generate_code(self):
```

Comment:
maybe we should document the return types. they are not quite clear to me. in the below code, sometimes the underscore version and sometimes the non-underscore version returns a tuple.

Reply:
good point, i will look into that. I was going to refactor the naming of these `generate_` and `export_` methods anyway to be more consistent for all types of components.

Reply:
Refactored the naming and added type hints